avocado/core/nrunner.py

```diff
@@ -928,10 +934,9 @@ def are_requirements_available(self, runners_registry=None):
             runners_registry = RUNNERS_REGISTRY_STANDALONE_EXECUTABLE
         return self.runnable.pick_runner_command(runners_registry)
 
-    def setup_output_dir(self):
-        self.output_dir = tempfile.mkdtemp(prefix='.avocado-task-')
-        env_var = {'AVOCADO_TEST_OUTPUTDIR': self.output_dir}
-        self.runnable.kwargs.update(env_var)
+    def setup_output_dir(self, output_dir):
+        self.output_dir = output_dir or tempfile.mkdtemp(prefix='.avocado-task-')
+        self.runnable.output_dir = self.output_dir
```

Comment:
I put it to the Runnable because it is needed on runner side and I leave it in Task because it was part of changes for the  https://avocado-framework.readthedocs.io/en/latest/guides/writer/chapters/writing.html#environment-variables-for-tests. But as @willianrampazzo pointed out, this PR breaks this feature, so I will have to make some changes here.